wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package component.javaparser;
+
+import org.custommonkey.xmlunit.DetailedDiff;
+import org.custommonkey.xmlunit.Diff;
+import org.custommonkey.xmlunit.ElementNameAndAttributeQualifier;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+/**
+ *This class will compare the previous component.xml file and latest component.xml file
+ */
+public class CompareXMLDiff {
+
+    private int numberOfReferenceTag;
+    private List<String> fileDifferences = new ArrayList<>();
+
+    public void compareXML(String file1, String file2) throws ParserConfigurationException, IOException, SAXException {
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        DocumentBuilder db = null;
+        Document doc = null;
+        Document doc2 = null;
+            db = dbf.newDocumentBuilder();
+            doc = db.parse(new File(file1));
```

Comment:
Please format the above code. Checkstyle plugin can be run from IDE against the modified source files to find formatting issues in your code.

Reply:
fixed checkstyle.